### ISSUE TITLE
Cannot narrow class or class instance from first argument of method

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1203]
**Describe the bug**
Classes (types) and class instances (objects) cannot be narrowed with `issubclass` and `isinstance`, respectively, from the first argument of a class or instance method.  In the first case, the type is "Unknown", and in the latter, it's `object`.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That `other` on line 7 would be `type[Foo]`; that `other` on line 11 would be `Foo`.

**Screenshots or Code**

```python
# pyright: strict

class Foo:
    @classmethod
    def bar(cls, other: type):
        if issubclass(other, cls):
            reveal_type(other)  # Type of "other" is "Unknown"

    def baz(self, other: object):
        if isinstance(other, self.__class__):
            reveal_type(other)  # Type of "other" is "object"

```

**VS Code extension or command-line**
Either, 1.1.88

**Comment by erictraut** [comment]
This will be implemented in the next release.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.89, which I just published. It will also appear in the next Pylance release.
